utils/DeployUtils.ps1

```diff
@@ -0,0 +1,240 @@
+        Creates a new private package repository (i.e., gallery) on local file system
+        .Parameter GalleryPath
+        Path for directory to use for private gallery
+        .Parameter GalleryName
+        Name of the private gallery
+        .Parameter Trusted
+        Indicates if private gallery is registered as a trusted gallery
+        .Example
+        New-PrivateGallery -Trusted
+        Create new private, trusted gallery using default name and location
+    #>
+    param (
+        [Parameter(Mandatory=$false)]
+        [ValidateScript({Test-Path -Path $_ -IsValid})]
+        [string]
+        $GalleryRootPath = $env:TEMP,
+        [Parameter(Mandatory=$false)]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $GalleryName = 'PrivateScubaGearGallery',
+        [switch]
+        $Trusted
+    )
+
+    $GalleryPath = Join-Path -Path $GalleryRootPath -ChildPath $GalleryName
+    if (Test-Path $GalleryPath){
+        Write-Debug "Removing private gallery at $GalleryPath"
+        Remove-Item -Recursive -Force $GalleryPath
+    }
+
+    New-Item -Path $GalleryPath -ItemType Directory
+
+    if (-not (IsRegistered -RepoName $GalleryName)){
+        Write-Debug "Attempting to register $GalleryName repository"
+
+        $Splat = @{
+            Name = $GalleryName
+            SourceLocation = $GalleryPath
+            PublishLocation = $GalleryPath
+            InstallationPolicy = if ($Trusted) {'Trusted'} else {'Untrusted'}
+        }
+
+        Register-PSRepository @Splat
+    }
+    else {
+        Write-Warning "$GalleryName is already registered. You can unregister: `nUnregister-PSRepository -Name $GalleryName"
+    }
+}
+
+function Publish-ScubaGearModule{
+    <#
+        .Description
+        Publish ScubaGear module to private package repository
+        .Parameter ModulePath
+        Path to module root directory
+        .Parameter GalleryName
+        Name of the private package repository (i.e., gallery)
+        .Parameter OverrideModuleVersion
+        Optional module version.  If provided it will use as module version. Otherwise, the current version from the manifest with a revision number is added instead.
+    #>
+    param (
+        [Parameter(Mandatory=$true)]
+        [ValidateScript({Test-Path -Path $_ -PathType Container})]
+        [string]
+        $ModulePath,
+        [Parameter(Mandatory=$false)]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $GalleryName = 'PrivateScubaGearGallery',
+        [Parameter(Mandatory=$false)]
+        [AllowEmptyString()]
+        [string]
+        $OverrideModuleVersion = ""
+    )
+
+    $ModuleBuildPath = Build-ScubaModule -ModulePath $ModulePath -OverrideModuleVersion $OverrideModuleVersion
+
+    if (SignScubaGearModule -ModulePath $ModuleBuildPath){
+        Publish-Module -Path $ModuleBuildPath -Repository $GalleryName
+    }
+    else {
+        Write-Error "Failed to sign module."
+    }
+}
+
+function Build-ScubaModule{
```

Comment:
I would prefer to see a description for internal helpful functions (see here and others below), just as there are descriptions for the public facing functions (like 'New-PrivateGallery'). Whenever someone else takes over this code, simple comments like this will be helpful to them.